configuration/download_cache.rst

```diff
@@ -0,0 +1,82 @@
+.. _download_cache:
+
+Download cache
+==============
+
+.. warning::
+
+    This is an **experimental** feature subject to breaking changes in future releases.
+
+.. warning::
+
+    This is an **advanced** feature, use it only if you fully understand this document. Do not use it if you are mixing api V1 (without revisions)
+    and api V2 (with revisions) clients.
+
+
+
+Conan implements a shared download cache that can be used to reduce the time needed to populate the Conan package cache
+with commands like :command:`install`, :command:`create`.
+
+This cache is purely an optimization mechanism. It is completely different to the Conan package cache, (typically the ``<userhome>/.conan`` folder).
+
+This cache will store in a given path, which can be configured in **conan.conf** file, the following:
+
+- All files that are downloaded from a Conan server (conan_server, Artifactory), both in the api V1 (without revisions) and V2 (with revisions).
+  This include files like *conanfile.py*, but also the zipped artifacts like *conan_package.tgz* or *conan_sources.tgz*
+- The downloads done by users with the ``tools.download()`` or ``tools.get()`` helpers, as long as they provide a checksum (md5, sha1, etc.). If
+  a checksum is not provided, even if the download cache is enabled, the download will be always executed.
+
+The cache uses the URL of the download, appending the checksum when provided. In the api V2, it is not necessary to append the checksum, because
+the URL itself already encodes the recipe revision and/or the package revisions, which are already checksums of the recipe and package respectively.
+
+.. warning::
+
+    The download cache will not be able to correctly cache artifacts with revisions enabled if those artifacts are created and uploaded repeatedly
+    with a client without revisions, because that will keep overwriting the revision "0".
```

Comment:
I would move the content of this warning together with the one at the beginning of the section, the are about the same: do not mix v1 and v2 and this cache

I wonder if we should detect _revision 0_ in the Conan v2 API and not use the cache for those artifacts...

Reply:
Yes, we should, will open a ticket to implement this protection in next releases. While writing it I realized that the docs might not be enough, and eventually users will hit this.

Reply:
https://github.com/conan-io/conan/issues/6467